Allow additionalRulesDirs to be specified in config

andrewnicols opened the pull request:
Allow a list of additionalRulesDirs to be specified in the
gherkin-lintrc configuration.

If a list of paths is provided as an argument then this takes precedence
over the value in configuration.

Any additionalRulesDirs specified in configuration is calculated
relative to the gherkin-lintrc.

andrewnicols commented:
> 4 unchanged lines in 2 files lost coverage.

These lines were already uncovered, and have just moved within the file. Not sure why they're showing as having lost coverage as they weren't covered before.

andrewnicols commented:
Pinging @vsiakka,

Any chance of a review of this code?

Thanks

vsiakka commented:
Hey @andrewnicols , thank for the change and sorry for the huge delay for my review. 
I really like the approach of being able to configure the additional rules dir in the configuration file. It makes sense for a team that wants to share the same additional rules, it makes it easy to call gherkin lint repeatedly with the same configuration. 

Having said that, I think treading "additionalRulesDirs" as another rule configuration is ideal in my opinion. I don't find it intuitive from the perspective of someone writing a config file or from the person working on the gherkin-lint code.

I think the correct approach for this is to change the layout of the `.gherkin-lintrc` files to be an object of:
```
"rules": {...},
"additionalRuleDirs": []
```
This will mean that the api of the project will change so it will be a major version release which I'm fine with doing.

andrewnicols commented:
Hi @vsiakka ,

Apologies for the delay on this one. I was in a car accident around this time, then had a child, and life happened.

I've made the following changes:
- rebased
- moved rules to `config.rules`
- move `additionalRulesDirs` to `config.additionalRulesDirs`
- updated all tests accordingly

andrewnicols commented:
I'm guessing this probably needs to be converted to using GHA rather than Travis. Let me know if there's anything I can do to help.

Cheers

andrewnicols commented:
Blocked by #248

andrewnicols commented:
Updated following feedback :)
